File: TestTask/Core/Entities/Product/Product.cs
using Core.Entities.Order;

namespace Core.Entities.Product;

public class Product : BaseEntity
{
    public Product()
    {
    }

    public Product(string name, int categoryId, string description) 
    {
        Name = name;
        CategoryId = categoryId;
        Description = description;
    }

    public string Name { get; set; }
    public ProductCategory Category { get; set; }
    public int CategoryId { get; set; }
    public string Description { get; set; }
    public DateTime CreatedDate { get; set; } = DateTime.UtcNow;
    public ICollection<ProductItem> ProductItems { get; set; }
    public ICollection<OrderItem> OrderItems { get; set; }
}

File: TestTask/frontend/src/app/models/product.ts
export interface ProductItem {
  id: number;
  productName: string;
  sizeId: number;
  size: string | undefined;
  categoryId: number;
  category: string | undefined;
  availableQuantity: number;
  createdDate: string;
  description: string;
  price: string;
}

export interface ProductSize {
  id: number;
  name: string;
}

export interface ProductCategory {
  id: number;
  name: string;
}

export interface Product {
  id: number;
  name: string;
  category: string;
  description: string;
  createdDate: string;
}
